addons/mrp/models/stock_move.py

```diff
@@ -379,6 +379,11 @@ def _should_bypass_set_qty_producing(self):
             return True
         if self.has_tracking != 'none' or self.state == 'done':
             return True
+        # Don't update qty_done when already manually changed
+        # if self._context.get('check_manual_change') and float_compare(self._origin.should_consume_qty, self.quantity_done, precision_rounding=self.product_uom.rounding):
+        #     return True
+        if float_compare(self._origin.should_consume_qty, self.quantity_done, precision_rounding=self.product_uom.rounding):
```

Comment:
Don't you think it could be more reliable if you compare to `self._origin.quantity_done` instead ?
Imagine you have a BoM with a component consumption to 1, you create a MO for this BoM with 10 qty to produce. So you have:
> Quantity: 0/10 To Produce
> Component:
> -- 0/10 To Consume 
> -- 0 Consumed

You change Qty Producing to 4 and save, you have now:
> Quantity: 4/10 To Produce
> Component:
> -- 4/10 To Consume 
> -- 4 Consumed (updated because `quantity_done` and `self._origin.should_consume_qty` were both 0)

But now, currently, if you change two times the Qty Producing (to 5 then to 7), you'll have this:
> Quantity: 5/10 To Produce
> Component:
> -- 5/10 To Consume 
> -- 5 Consumed (updated because `quantity_done` and `self._origin.should_consume_qty` were both 4)

> Quantity: 7/10 To Produce
> Component:
> -- 7/10 To Consume 
> -- **5 Consumed** (stay at 5 because `quantity_done` was 5 and `self._origin.should_consume_qty` was still 4 as there was no save between the two changes)

By comparing with the `_origin.quantity_done`, it will update the consumed qty even if you manually changed it if you didn't save between, but I think we can't have the good side of the two ways. Maybe it should be clarified with @mathieu-mgm what he prefer ?